### PR TITLE
Adding new list (~1440) of Arxiv papers and code for rescraping with ar5iv

### PR DESCRIPTION
**Pull request by jknowak**
- fixed some bug with file write (.txt)
- changed method in arxiv scraper to do more and try ar5iv.org as a backup
- imported bs4 to work with html
- used the output of last scrape as a base of future arxiv scrape when it comes to paper ids

### CODE REVIEW

File: align_data/arxiv_papers/arxiv_papers.py [hunk 1]
```diff
@@ -94,50 +98,104 @@ def fetch_entries(self) -> None:
...
+        if "//arxiv.org" in r.url:
             return None
+        try:
+            soup = BeautifulSoup(r.content, features="xml")
+        except Exception as e:
+            logger.error(e)
+            return None
+        if not self._is_bad_soup(soup,parser=parser):
+            return self._article_markdown_from_soup(soup)
+        else:
```
Comment (mruwnik) [inline align_data/arxiv_papers/arxiv_papers.py:166]: remove the else

File: align_data/arxiv_papers/arxiv_papers.py [hunk 2]
```diff
@@ -94,50 +98,104 @@ def fetch_entries(self) -> None:
...
+        Get arxiv link
+        """
+        return f"https://arxiv.org/abs/{paper_id}"
 
+    def _remove_bib_from_article_soup(self, article_soup) -> str:
+        """
+        Strip markdown
+        """
+        bib = article_soup.find("section", id="bib")
+        if bib is None:
```
Comment (mruwnik) [inline align_data/arxiv_papers/arxiv_papers.py:181]: how about moving this around to:
```
if bib:
    bib.decompose()
return article_soup
```?

File: requirements.txt [hunk 3]
```diff
@@ -3,6 +3,7 @@ tqdm
 jsonlines
 path
 requests
+beautifulsoup4
```
Comment (mruwnik) [inline requirements.txt:6]: is this needed?

File: align_data/arxiv_papers/arxiv_papers.py [hunk 4]
```diff
@@ -94,50 +98,104 @@ def fetch_entries(self) -> None:
...
+        logger.info(f"Fetching {link}")
         try:
-            r = requests.get(v_link, timeout=5 * self.COOLDOWN)
+            r = requests.get(link, timeout=5 * self.COOLDOWN)
         except Exception as e:
             logger.error(e)
             return None
-        markdown = markdownify(r.content)
-        if self._is_dud(markdown):
+        if "//arxiv.org" in r.url:
```
[the remark this answers is not in the document]
Comment (jknowak) [reply align_data/arxiv_papers/arxiv_papers.py:157]: yes, but it also redirects to itself so I didn't want to turn off all redirects

File: requirements.txt (hunk 3 above)
Comment (jknowak) [reply requirements.txt:6]: yes, that's what I use to parse html into BeautifulSoup() to easily find specific html tags and pick them (like choose only <article>) or .decompose them (to get rid of <script>).

File: align_data/arxiv_papers/arxiv_papers.py [hunk 5]
```diff
@@ -94,50 +98,104 @@ def fetch_entries(self) -> None:
...
-        """
-        Get arxiv link
-        """
-        return f"https://arxiv.org/abs/{paper_id}"
+    def _is_bad_soup(self, soup, parser='vanity') -> bool:
+        if parser == 'vanity':
+            vanity_wrapper = soup.find("div", class_="arxiv-vanity-wrapper")
+            if vanity_wrapper is None:
+                return None
+            else:
```
[the remark this answers is not in the document]
Comment (jknowak) [reply align_data/arxiv_papers/arxiv_papers.py:106]: ah, yes, bc of the return

File: requirements.txt (hunk 3 above)
Comment (mruwnik) [reply requirements.txt:6]: yes, but isn't it already installed as a requirement of something else? BeautifulSoup is already used in other places in the code

Comment (jknowak) [reply requirements.txt]: you're right, it is installed with gdown, removed it

File: align_data/arxiv_papers/arxiv_papers.py [hunk 6]
```diff
@@ -33,7 +34,7 @@ def _get_arxiv_metadata(self, paper_id) -> arxiv.Result:
 
     @property
     def items_list(self):
-        self.papers_csv_path = self.raw_data_path / "ai-alignment-papers.csv"
+        self.papers_csv_path = self.raw_data_path / "ai-alignment-arxiv-papers-new.csv"
```
Comment (mruwnik) [inline align_data/arxiv_papers/arxiv_papers.py:37]: I'd remove the `new` - git has a proper history so it should be possible to recover any lost files. If you want to keep 2 files, I'd recommend renaming the previous one to `ai-aligment-arxiv-papers-old.csv` and having this class always point to the current one

**Review by mruwnik — APPROVED** [review]

File: align_data/arxiv_papers/arxiv_papers.py (hunk 6 above)
Comment (jknowak) [reply align_data/arxiv_papers/arxiv_papers.py]: Done

**Comment by jknowak** [comment]
Who is doing the merge @mruwnik ?

**Review by mruwnik — APPROVED** [review]

**Comment by mruwnik** [comment]
In general it seems to be good form for the author to press the final merge. Unless you don't have permissions, of course. So when the button is green and you think it's mergable, feel free to do so

**Comment by mruwnik** [comment]
one thing, though - could you change the merge method to `Squash and merge` on the merge button? Otherwise the git history quickly gets useless

**Pull request merged**